lib/course_api.rb

```diff
@@ -0,0 +1,153 @@
+require 'terms'
+require 'json'
+require 'nokogiri'
+require 'faraday'
+
+# Get course term information and then parse
+class CourseAPI
+    def initialize(xml)
+        @course_term = Nokogiri::XML(xml)
+    end
+    # Parse the XML returned by the CourseTerm MaIS API
+    def parse
+        courses = []
+        all_errors = []
+        # Return array of course hash objects
+        courses_list = parse_courses()
+        counter = 0
+        courses_list.each do |course|
```

Comment:
Could we do away with the internal counter if we use `each_slice` instead?

```
courses_list.each_slice(10000) do |courses|
  courses.each do |course|
     ...
  end

  sleep(1)
end
```

or, I suppose, use each_with_index and `index % 10000`?